server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -14,30 +15,42 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.regex.Matcher; 
+import java.util.regex.Pattern; 
 
 @WebServlet("/api/analysis")
 public class AnalysisServlet extends HttpServlet {
   @Override
   protected void doGet(HttpServletRequest req, HttpServletResponse res)
       throws ServletException, IOException {
 
-    String videoName = req.getParameter("name");
-
-    assert videoName == null : "Something went wrong with sending to backend.";
+    String userInput = req.getParameter("name");
 
     // Use like this: {url_parameter, value}
     HashMap<String, String> videoArgs = new HashMap<>();
     HashMap<String, String> commentArgs = new HashMap<>();
     HashMap<String, String> nameArgs = new HashMap<>();
 
-    videoArgs.put("q", videoName);
-    videoArgs.put("type", "video");
-    String videoIdJson = new YoutubeRequest("search", videoArgs).getResult();
-    String videoId = getVideoId(videoIdJson);
+    String videoId = userInput; // assume user enters id
 
     commentArgs.put("part", "snippet");
-    commentArgs.put("videoId", videoId);
-    String commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    commentArgs.put("videoId", userInput);
+    String commentsJson;
+    
+    // Test if its a youtube id from the beginning
+    if (userInput.length() == 11 && !thereIsWhiteSpace(userInput)) {
+        try {
+            commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+        } catch (IOException err) {
+            videoId = getFirstVideoFromSearch(userInput);
+            commentArgs.replace("videoId", videoId);
+            commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
```

Comment:
nit: this code is identical to what's in your `else` clause. You can just use an if statement where you check if commentsJson is null after trying to get comments with the video id, and only leave a comment in your catch
```
if (/** is a video id */) {
  try {
    commentsJson = getVideoComments(videoId);
  } catch {
    // It wasn't a video id.
  }
}

if (!commentsJson) {
  // Search video and get comments.
}
```